### PR TITLE
#4658 Added feature switch to have ability to have only 1 review per product from customer

### PR DESCRIPTION
**Pull request by denisburlacu**

### CODE REVIEW

File: src/Presentation/Nop.Web/App_Data/Localization/defaultResources.nopres.xml [hunk 1]
```diff
@@ -5526,9 +5526,15 @@
   <LocaleResource Name="Admin.Configuration.Settings.Catalog.ProductReviewsMustBeApproved.Hint">
     <Value>Check if product reviews must be approved by administrator.</Value>
   </LocaleResource>
-  <LocaleResource Name="Admin.Configuration.Settings.Catalog.ProductReviewsPageSizeOnAccountPage">
-    <Value>Product reviews page size</Value>
-  </LocaleResource>
+    <LocaleResource Name="Admin.Configuration.Settings.Catalog.OneReviewPerProductFromCustomer">
+        <Value>One review per product from customer</Value>
```
Comment (denisburlacu) [inline src/Presentation/Nop.Web/App_Data/Localization/defaultResources.nopres.xml:5530]: any helps in adjusting Localizations will be great

File: src/Libraries/Nop.Services/Installation/CodeFirstInstallationService.cs [hunk 2]
```diff
@@ -6254,6 +6254,7 @@ protected virtual void InstallSettings()
                 ShowShareButton = true,
                 PageShareCode = "<!-- AddThis Button BEGIN --><div class=\"addthis_toolbox addthis_default_style \"><a class=\"addthis_button_preferred_1\"></a><a class=\"addthis_button_preferred_2\"></a><a class=\"addthis_button_preferred_3\"></a><a class=\"addthis_button_preferred_4\"></a><a class=\"addthis_button_compact\"></a><a class=\"addthis_counter addthis_bubble_style\"></a></div><script src=\"http://s7.addthis.com/js/250/addthis_widget.js#pubid=nopsolutions\"></script><!-- AddThis Button END -->",
                 ProductReviewsMustBeApproved = false,
+                OneReviewPerProductFromCustomer = false,
```
Comment (denisburlacu) [inline src/Libraries/Nop.Services/Installation/CodeFirstInstallationService.cs:6257]: actually this is not needed as a boolean by default is having false value, but for keeping consistency added default

File: src/Libraries/Nop.Services/Catalog/ProductService.cs [hunk 3]
```diff
@@ -2296,7 +2296,7 @@ public virtual ProductReview GetProductReviewById(int productReviewId)
         /// </summary>
         /// <param name="productReviewIds">Product review identifiers</param>
         /// <returns>Product reviews</returns>
-        public virtual IList<ProductReview> GetProducReviewsByIds(int[] productReviewIds)
```
Comment (denisburlacu) [inline src/Libraries/Nop.Services/Catalog/ProductService.cs:2299]: sorted some typo

File: src/Presentation/Nop.Web/Infrastructure/Permissions/ProductReview/IProductReviewPermissionService.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using Nop.Core.Domain.Catalog;
+using Nop.Core.Domain.Customers;
+using Nop.Core.Domain.Stores;
+
+namespace Nop.Web.Infrastructure.Permissions.ProductReview
+{
+    public interface IProductReviewPermissionService
```
Comment (denisburlacu) [inline src/Presentation/Nop.Web/Infrastructure/Permissions/ProductReview/IProductReviewPermissionService.cs:7]: i added this as a permission service, actually the name is not good, will be great if you can share a proper name im thinking also about IProductReviewPermissionHandler
-
at the moment i didnt saw a way to have some places to be able to share this code, as i cant move this to Nop.Services as here i use some caching staff, and CacheKey creator is in Nop.Web, not sure what approach is taking right now for this kind of things, but will appreciate if we can have a chat on this

**Comment by denisburlacu** [comment]
@AndreiMaz, @RomanovM  can we please assign someone to review this please?

File: src/Presentation/Nop.Web/Infrastructure/Permissions/ProductReview/IProductReviewPermissionService.cs (hunk 4 above)
Comment (iammourya) [reply src/Presentation/Nop.Web/Infrastructure/Permissions/ProductReview/IProductReviewPermissionService.cs:7]: denisburlac - how about naming it as "IProductReview**Restriction**Service.cs"

**Comment by DmitriyKulagin** [comment]
@denisburlacu Thanks for the suggestion, I made the necessary adjustments to the solution to avoid conflicts with the current develop branch.

**Pull request merged**